internal/pkg/agent/transpiler/ast_test.go

```diff
@@ -1071,6 +1073,29 @@ func TestLookup(t *testing.T) {
 	}
 }
 
+func TestCondition(t *testing.T) {
+	vars := mustMakeVars(map[string]interface{}{
+		"other": map[string]interface{}{
+			"data": "info",
+		}})
+
+	input := NewKey("condition", NewStrVal("${other.data} == 'info'"))
```

Comment:
There should be a test for the case where the condition is edited for the same input.

As an implementation question does the cache need to be invalidated when this happens, or is a new AST node created so that we don't need to care?

Reply:
The ability to modify an AST is very limited to begin with. The only public API to do it is the `Insert` function, which only works on Dicts and Lists. I can add a test where I replace a key in a dict and verify the condition doesn't get saved.

Reply:
https://github.com/elastic/elastic-agent/pull/6229/commits/b5ac1c4176c7f423468bf191508f527fc81a2240